oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/resources/RmUtil.java

```diff
@@ -56,7 +59,12 @@ public static ResourceShape lookupRequirementsInstanceShapes(final String servic
 									URI[] instanceShapes = creationFactory.getResourceShapes();
 									if (instanceShapes != null ){
 										for ( URI typeURI : instanceShapes) {
-											response = client.getResource(typeURI.toString(),OSLCConstants.CT_RDF);
+											if(configurationContext != null) {
+												response = client.getResource(typeURI.toString(), null, OSLCConstants.CT_RDF, configurationContext);
+											}
+											else{
```

Comment:
Do we need this? I thought your code handles null as a missing context.

Reply:
You're right. This is a redundant check.